output.tf

```diff
@@ -1,11 +1,7 @@
-output "security_group_id" {
-  value = aws_security_group.elasticache.id
-}
-
-output "endpoints" {
-  value = aws_elasticache_cluster.default.cache_nodes.*.address
+output "primary_endpoints" {
+  value = var.replication_group ? aws_elasticache_replication_group.default[0].primary_endpoint_address : aws_elasticache_cluster.default[0].cache_nodes.*.address
 }
 
-output "ports" {
-  value = aws_elasticache_cluster.default.cache_nodes.*.port
+output "security_group_id" {
+  value = aws_security_group.elasticache.id
 }
```

Comment:
Fully agree that if something is an input, it should not be an output. But maybe this is now used in other configs, so I prefer to keep it in here.